webpack.config.js

```diff
@@ -0,0 +1,54 @@
+const path = require('path');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const ESLintPlugin = require('eslint-webpack-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
+  entry: {
+    'codec-string': path.resolve(__dirname, './src/decode.js'),
+    app: {
+      dependOn: 'codec-string',
+      import: path.resolve(__dirname, './src/app.js'),
+    },
+  },
+  module: {
+    rules: [
+      {
+        test: /\.js$/,
+        exclude: /node_modules/,
+        use: ['babel-loader'],
+      },
+      {
+        test: /\.ts$/,
+        exclude: /node_modules/,
+        use: ['ts-loader'],
+      },
+      {
+        test: /\.css$/i,
+        use: ["style-loader", "css-loader"],
+      }
+    ]
+  },
+  resolve: {
+    extensions: ['*', '.js']
+  },
+  plugins: [
+    new CleanWebpackPlugin(),
+    new ESLintPlugin(),
+    new HtmlWebpackPlugin({
+      title: 'Codec string decoder',
+    })
+  ],
+  output: {
+    path: path.resolve(__dirname, './dist'),
+    filename: '[name].min.js'
+  },
+  devServer: {
+    static: path.resolve(__dirname, './dist'),
+    allowedHosts: [
+      '.dev.youview.co.uk',
```

Comment:
Might want to remove this and replace from a dynamically read file, e.g. a `.env` file.